addon/components/inputs/datetime.js

```diff
@@ -13,12 +14,40 @@ export default AbstractInput.extend({
 
   layout,
 
+  // == State Properties =======================================================
+
+  getDefaultProps () {
+    if (this.get('value')) {
+      var arr = this.get('value').split('T')
+
+      return {
+        date: arr[0] || '',
+        time: arr[1] || ''
+      }
+    }
+
+    return {
+      date: '',
+      time: ''
+    }
+  },
+
   // == Computed Properties ====================================================
 
   // == Functions ==============================================================
 
+  parseValue (value) {
+    if (value.includes('-')) {
+      this.set('date', value)
+    } else {
+      this.set('time', 'T' + value)
+    }
+    return this.get('date') + this.get('time')
+  },
```

Comment:
With the computed property change this can simply become:
```js
parseValue (value) {
  if (value.includes('-') {
    return `${this.get('date')}T${value}`
  }

  return `${value}T${this.get('time')}`
}
```